packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -54,33 +56,35 @@ export async function getFileLink(
                 result = await getFileLinkCore(getToken, odspUrlParts, identityType, logger);
                 success = true;
             } catch (err) {
-                // If it is not retriable, then just return undefined
+                // If it is not retriable, then just throw
                 if (!canRetryOnError(err)) {
-                    fileLinkCache.remove(cacheKey);
-                    return undefined;
+                    // Delete from the cache to permit retrying later.
+                    fileLinkCache.delete(cacheKey);
```

Comment:
I think if we get here, there is no entry in the cache, no?

Reply:
Here it should be in the cache in pending state (for the purpose that simultaneous callers will not doubly-invoke), since we set the promise in the cache before awaiting the result.

Reply:
But I think you killed that by moving to use map :) Which actually raises an interesting question - what happens if there are two calls in parallel? If the old code handled it, then maybe we should keep using PromiseCache?

Reply:
This is functionally equivalent to the meat of `PromiseCache`.  Both this implementation and `PromiseCache` invoke the promise-generating function and store the returned `Promise` in a `Map`.

From [PromiseCache](https://github.com/microsoft/FluidFramework/blob/main/common/lib/common-utils/src/promiseCache.ts#L146):

```ts
// Start the async work and put the Promise in the cache
promise = safeAsyncFn();
this.cache.set(key, promise);
```

vs. here:
```ts
const fileLink = valueGenerator();
fileLinkCache.set(cacheKey, fileLink);
```

In this implementation we early-out if `maybeFileLinkCacheEntry` is not `undefined` above, preventing parallel execution.  There should be no regression.